linode/instanceip/framework_resource.go

```diff
@@ -54,13 +54,59 @@ func (r *Resource) Create(
 	isPublic := plan.Public.ValueBool()
 
 	client := r.Meta.Client
-	ip, err := client.AddInstanceIPAddress(ctx, linodeID, isPublic)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			fmt.Sprintf("Failed to created instance (%d) IP", linodeID),
-			err.Error(),
-		)
-		return
+	var ip *linodego.InstanceIP
```

Comment:
Given reserved IPs use a separate set of endpoints (`networking/reserved/ips`), I wonder if it'd be worth creating a new `linode_instance_reserved_ip` resource entirely 🤔 

Given there is a lot of overlap, something like a `reserved` bool field might also suffice.

Reply:
I already have another PR open for the new resource called `networkreservedips`. I kept these functionalities here to maintain consistency with the linodego library.

Yes I have added the reserved bool field in the model of the `networkreservedips` resource.

Reply:
`networkreservedips` resource handles 4 other endpoints - Create, Fetch, List, Delete reserved IP addresses.

Reply:
Since we have a separate resource to manage the reservedipds, I'm wondering if it might be better to not having them here to keep this instanceip resource staightforward.

Reply:
The reason I kept the functionality in the `instanceip` resource is because the linodego function to assign reserved IP to instances is within the `instance_ips.go` class of linodego. So I thought it would improve consistency. (The reason the function is in the `instance_ips.go` is because the endpoint matches the others in the class)

Reply:
Sorry for the late response to this thread! 

I think I'm with @yec-akamai here, it might make more sense to exclude this logic from the `linode_instance_ip` resource since this resource specifically corresponds to a single non-reserved IP 🙂